### DISCUSSION TITLE
Please, make Git Extensions a 64-bit application

### DISCUSSION THREAD

**Discussion by iassenev** [gitextensions/gitextensions#9869] [Ideas]
On large repositories, GE fails with OutOfMemory exception with commit size ~= 2GiB
One may link it with /LARGEADDRESSAWARE option to get access to the higher 2GiB, but it would be more reasonable to make GE a 64-bit application in the year 2022. There are no 32-bit CPUs at all, Intel Atoms are a history.

**Comment by mdonatas** [comment]
Hmm.. quite a few assumptions and unnecessary jabs _in the year 2022_ in your request.

I've managed to get an OOM exception but it has nothing to do with 32 vs 64 bits. It's because there's an attempt to load the file contents and default dotnet data structures support max size of 2GB. One needs to employ some smarts to overcome this but that requires work.

Is that where you get this error?
![image](https://user-images.githubusercontent.com/483659/155477792-3196fd79-48c6-4683-9763-61f5d4bb37c6.png)

**Comment by RussKie** [comment]
Yes, it is year 2022. It is also an FOSS project. 
You're welcome to submit a PR with a change.
